src/main/java/com/teragrep/akv_01/event/ParsedEventListFactory.java

```diff
@@ -76,15 +98,43 @@ public List<ParsedEvent> asList() {
         final List<ParsedEvent> events = new ArrayList<>(payloads.length);
         for (int i = 0; i < payloads.length; i++) {
             if (payloads[i] != null) {
+                EventPartitionContext eventPartitionContext = eventPartitionContextStub;
+                EventProperties eventProperties = eventPropertiesStub;
+                EventSystemProperties eventSystemProperties = eventSystemPropertiesStub;
+                EnqueuedTime enqueuedTime = enqueuedTimeStub;
+                EventOffset eventOffset = eventOffsetStub;
+
+                if (partitionCtx != null) {
```

Comment:
it would be better to use 
```
final EventPartitionContext partitionCtx;
if (partitionCtx != null) {
 partitionCtx = ...
}
else {
partitionCtx = stub
}